src/components/Experience.ts

```diff
@@ -36,4 +36,76 @@ export class Experience<TGameStartr extends FullScreenPokemon> extends Component
             this.levelup(pokemon);
         }
     }
+
+    /**
+     * Calculates how much experience a defeated Pokemon is worth.
+     * https://bulbapedia.bulbagarden.net/wiki/Experience#Gain_formula
+     *
+     * @param pokemon   The defeated Pokemon.
+     * @param trainer   If the defeated Pokemon had a trainer.
+     */
+    public calculateExperience(pokemon: IPokemon, trainer: boolean): number {
+        const participatedPokemon: IPokemon[] = this.gameStarter.itemsHolder.getItem("battleParticipants");
+        const numerator: number = this.calculateNumerator(pokemon.level, trainer,
+                                                          this.gameStarter.constants.pokemon.byName[pokemon.title.join("")].experience);
+
+        if ("EXP ALL" in this.gameStarter.itemsHolder.getItem("items")) {
```

Comment:
I don't like hardcoding. This assumes that "EXP ALL" is the only possible item that can modify experience settings. There are actually plenty of ways that get introduced, such as EXP SHARE and bonuses for traded Pokemon. It's also likely that someone will want to write a mod to change experience setting.

A more sustainable approach would be to allow items to have a method called when experience is calculated if their Pokemon was involved in the battle.